docs/DEVELOPMENT.md

```diff
@@ -497,3 +501,54 @@ Summary table of "types" (the first part of the commit message). More details at
 | ci       | Commits to trigger CI actions but no version changes      | none         |                   |
 
 Major version bumps are created when `BREAKING CHANGE` is in the commit body or the type is followed by `!` (e.g. `feat!: add a new major feature`). All types can be modified with an additional "scope" as shown in the conventional commit documentation (e.g. `fix(admin): remove broken link in admin panel`).
+
+## Frontend conventions
+
+The following conventions used to build and maintain the frontend UI.
+
+*NOTE: At this time, the current state of the frontend structure and conventions do not follow this documentation. This document is setting standards moving forward.
+
+### Architecture
+
+The frontend UI should be structured to improve consistency and developer experience by establishing and maintaining the following standards. The frontend consists of a single page application architecture using Nunjucks templates to rendered by the server and React to render pages in the browser.
+
+#### Server rendered views (Nunjucks)
+- Templates in the [`./views`](../views) directory.
+
+#### Browser rendered pages (React)
+- Page template components are in the [`./frontend/pages`](../frontend/pages) directory.
+- Only page template components should be used to render the routes in the [`./frontend/routes.js`](../frontend/routes.js) file.
+- Pages are composed of two types of components: Pure components and dynamic components. The majority of components used should be written as pure components.
+- Most HTTP requests should be called from the page component with the response data being passed down to pure components. Dynamic components should be used sparingly and scoped to a specific purpose to encapsulate complexity and state to the specific dynamic component.
```

Comment:
How strict do we want to be about the separation? For example, can a dynamic component have any markup? Or should it just be a container for other pure components?

Reply:
I think we should opt for minimal markup in a dynamic component and it should be used sparingly when the markup is so specific to the dynamic component that it would never be used in any other circumstances. As we transition the frontend, we'll have a better understanding of our best practices.